### PR TITLE
[snarky] add canonical sort in permutation for better interop

### PR DESCRIPTION
**Pull request by mimoo**
Before this change, the order in which the kimchi permutation is created is strongly tied to the (albeit non-deterministic) implementation of the libraries used by the plonk_constraint_system.

This makes it extra hard to obtain the same permutation (and thus the same circuit digest) in a different implementation of snarky.

This change sorts all the cycles, to provide a more "canonical" permutation.

Concretely, we could have a cycle that would look like this before: 

```
(0, 1) -> (3, 2) -> (1, 5)
```

It will now always look like this:

```
(0, 1) -> (1, 5) -> (3, 2)
```

### CODE REVIEW

File: src/lib/crypto/snarky_tests/examples/ternary.json [hunk 1]
```diff
@@ -1 +1 @@
...
...0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,237,48,45,153,27,249,76,9,252,152,70,34,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,64],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,237,48,45,153,27,249,76,9,252,152,70,34,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,64],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]]},{"typ":"Generic","wires":[{"row":1,"col":2},{"row":2,"col":1},{"row":2,"col":2},{"row":2,"col":3},{"row":2,"col":4},{"row":2,"col":5},{"row":2,"col":6}],"coeffs":[[1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[255,255,255,255,236,48,45,153,27,249,76,9,252,152,70,34,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,64]]}]}
```
Comment (mimoo) [inline src/lib/crypto/snarky_tests/examples/ternary.json:1]:
```
row0.pub.Generic<1,0,0,0,0>
.l1 -> row1.l1

row1.Generic<1,0,-1,0,1><-1,0,0,1,0>
.l1 -> .l2, .o1 -> row2.l1
.l2 -> .r2, .r2 -> row0.l1

row2.Generic<1,0,0,0,-2>
.l1 -> row1.o1
```

File: src/lib/crypto/snarky_tests/examples/ternary.json [hunk 2]
```diff
@@ -1 +1 @@
...
```
[the remark this answers is not in the document]
Comment (mimoo) [reply src/lib/crypto/snarky_tests/examples/ternary.json:1]: (using https://github.com/o1-labs/proof-systems/pull/1043)

**Comment by mimoo** [comment]
!ci-build-me

**Comment by joseandro** [comment]
!ci-build-me

**Comment by mimoo** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
Done elsewhere. Closing.

**Pull request closed without merging**